packages/common/src/permissions/_internal/checkFeature.ts

```diff
@@ -0,0 +1,90 @@
+import { IArcGISContext } from "../../ArcGISContext";
+import { IPermissionPolicy, PolicyResponse, IPolicyCheck } from "../types";
+import { getPolicyResponseCode } from "./getPolicyResponseCode";
+import { HubEnvironment } from "../../ArcGISContext";
+import { env } from "process";
+
+const HubFeatures = ["workspaces"];
+
+export type HubFeature = (typeof HubFeatures)[number];
+
+// TODO: i don't think i have this right
+export type IHubFeatures = Record<HubFeature, boolean>;
+
+const QaExtFeatures: IHubFeatures = {
+  workspaces: true,
+};
+
+const DevExtFeatures: IHubFeatures = { ...QaExtFeatures };
+const LocalDevFeatures: IHubFeatures = { ...QaExtFeatures };
+const SandboxDevFeatures: IHubFeatures = { ...QaExtFeatures };
+
+const ProductionFeatures: IHubFeatures = {
+  workspaces: false,
+};
+
+const EnterpriseFeatures: IHubFeatures = {
+  workspaces: false,
+};
+
+function getFeaturesForEnvironment(environment: HubEnvironment): IHubFeatures {
+  switch (environment) {
+    case "qaext":
+      return QaExtFeatures;
+    case "devext":
+      return DevExtFeatures;
+    case "localdev":
+      return LocalDevFeatures;
+    case "sandbox":
+      return SandboxDevFeatures;
+    case "enterprise":
+      return EnterpriseFeatures;
+    default:
+      // default to prod
+      return ProductionFeatures;
+  }
+}
+
+function getFeatures(context: IArcGISContext): IHubFeatures {
+  const { environment } = context;
+  const featuresForEnvironment = getFeaturesForEnvironment(environment);
+  return {
+    ...featuresForEnvironment,
+    ...context.features,
+  };
+}
```

Comment:
Merges the features for the environment with any overrides we got on the URL queryparams.